code/backend/batch/utilities/search/search_handler_base.py

```diff
@@ -17,10 +17,16 @@ def search_with_facets(self, query: str, facets: list[str]):
             return None
         return self.search_client.search(query, facets=facets)
 
-    def get_unique_files(self, results, facet_key: str):
-        if results:
-            return [facet["value"] for facet in results.get_facets()[facet_key]]
-        return []
+    def get_unique_files(self, search_text, field_name):
+        unique_files = set()  # Using a set to automatically handle uniqueness
```

Comment:
Can we use a unit test to enforce this rather than a comment?

Reply:
Sure. Added test cases, can you please review it once